api/envoy/config/route/v3/route_components.proto

```diff
@@ -814,6 +814,9 @@ message RouteAction {
 
     // Determines if the trace span should be sampled. Defaults to true.
     google.protobuf.BoolValue trace_sampled = 4;
+
+    // Disables appending the ``-shadow`` suffix to the shadowed cluster name. Defaults to false.
+    google.protobuf.BoolValue disable_shadow_host_suffix_append = 6;
```

Comment:
OOC the field defaults to false, so why not to use plain `bool` type? It may make sense to use a wrapped type if it is expected that the default value will change in the future, but I'm not sure if this is the case.

Reply:
I used the well-known type to avoid a `false` getting marshaled for that field.

Reply:
What does getting marshaled refer to here?

Reply:
Whenever this proto is marshaled to anything... JSON, XML, etc.
If it's a raw `bool`, then the field will be explicitly marshaled... otherwise if it's not set, the `null` would be omitted?

Essentially I just used the WKT to avoid any side-effects.

Reply:
I don't think that's how proto3 -> JSON translation works by default, see https://protobuf.dev/programming-guides/proto3/#json:

> Always emit fields without presence: Fields that don’t support presence and that have their default value are omitted by default in JSON output (for example, an implicit presence integer with a 0 value, implicit presence string fields that are empty strings, and empty repeated and map fields). An implementation may provide an option to override this behavior and output fields with their default values.

Envoy API style prefers to use scalar for these cases.

Reply:
Huh.  
I was under the impression that WKTs were used so that they were omitted from the JSON when not set, and scalar fields are always serialized.

Reply:
No, WKTs are used to support the testing of absence / presence to provide the ability to provide default values for example.